swan-lake/learn-the-platform/run-in-the-cloud/function-as-a-service/azure-functions.md

```diff
@@ -214,4 +219,175 @@ Once the function is deployed, add an item to the collection.
 
 >**Note:** For a full sample with all the supported Azure Functions triggers and bindings in Ballerina, see the [Azure Functions deployment example](/learn/by-example/azure-functions-deployment).
 
+## Azure Functions Native (Experimental)
+
+### Prerequisites
+1. Install the latest Ballerina distribution.
+2. Install the <a href="https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest" target="_blank">Azure CLI</a>.
+3. Install and configure [Functions Core Tools](https://learn.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=v4%2Clinux%2Ccsharp%2Cportal%2Cbash#install-the-azure-functions-core-tools)
+4. Login to the Azure CLI by executing the `az login` command.
+5. <a href="https://www.docker.com/" target="_blank">Docker</a> installed and configured in your machine
+6. GraalVM [installed and configured](https://ballerina.io/learn/build-a-native-executable/#set-up-the-prerequisites) in your machine
+7. Create an <a href="https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-function-app-portal" target="_blank">Azure Function app</a> with the given resource group with following requirements.
+
+>**Note:** Make sure to remember the function application name and storage account name as they will be required in the code samples.
+   - Runtime stack - `Java 11`
+   - Hosting operating system - `Linux` (As of now, Ballerina Azure functions native support is only available for Linux.)
+
+### Write the function
+
+You can use any azure functions code in the native approach. We wil use the same code as in the [generic azure functions](#write-the-function) section for simplicity.
+
+Create a Ballerina package.
+```
+$ bal new azure_functions_native
+```
+Replace the contents of the generated BAL file with the following content.
+
+```ballerina
+import ballerinax/azure_functions as af;
+
+service / on new af:HttpListener() {
+    resource function get hello(string name) returns string {
+        return "Hello, " + name + "!";
+    }
+}
+```
+
+### Build and testing the function
+
+Azure functions package supports two build options. You can find the description about each build option below.
+* cloud="azure_functions" - Builds the native executable to be compatible with azure functions cloud.
+* cloud="azure_functions_local" - Builds the native executable to be compatible with your own machine for development purposes.
+
+>**Note:** For the non native JVM based approach, both build options will behave the same as JVM is platform independent.
+
+We will be using these build options along with native build option while building the package. First, let's build the package for running locally. This will use the GraalVM you installed in your machine to build the native image and the generated executable be compatible with your machine.
```

Comment:
```suggestion
This sample uses these build options along with the native build option while building the package. First, let's build the package and run it locally. This will use the GraalVM you installed on your machine to build the native image and make the generated executable compatible with your machine.
```